src/service/resources-impl.js

```diff
@@ -1454,15 +1454,19 @@ export class Resources {
           this.exec_.getSize() == 0 &&
           this.queue_.getSize() == 0 &&
           now > this.exec_.getLastDequeueTime() + 5000) {
-      let idleScheduledCount = 0;
+      let idleScheduledPreloadCount = 0;
+      let idleScheduledLayoutCount = 0;
       for (let i = 0; i < this.resources_.length; i++) {
         const r = this.resources_[i];
         if (r.getState() == ResourceState.READY_FOR_LAYOUT &&
                 !r.hasOwner() && r.isDisplayed()) {
-          dev().fine(TAG_, 'idle layout:', r.debugid);
-          this.scheduleLayoutOrPreload_(r, /* layout */ false);
-          idleScheduledCount++;
-          if (idleScheduledCount >= 4) {
+          const idleLayout = r.idleRenderOutsideViewport();
+          if ((idleLayout && idleScheduledLayoutCount++ < 4) ||
+            (!idleLayout && idleScheduledPreloadCount++ < 4)) {
```

Comment:
I’m still worried this will hog too much CLU time. What does just preload do again? /cc @dvoytenko